### PR TITLE
Task/135 Storing FCM token

### PR DESCRIPTION
**Pull request by rafalkrystek**

### CODE REVIEW

File: domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.safetyheads.akademiaandroida.domain.usecases
+
+
+import com.safetyheads.akademiaandroida.domain.repositories.UserRepository
+import kotlinx.coroutines.flow.Flow
+
+class UpdateProfileFCMUseCase(private val repository: UserRepository) :
+    BaseUseCase<UpdateProfileFCMUseCase.UpdateProfileParam, Unit> {
```
Comment (msarelo) [inline domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt:8]: You can use here `ParameterlessBaseUseCase`

File: domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+    data class UpdateProfileParam(val userUUID: String, val FCMToken: String) : BaseUseCase.Params
```
Comment (msarelo) [inline domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt:10]:
```suggestion
    data class UpdateProfileParam(val userUUID: String, val fcmToken: String) : BaseUseCase.Params
```

File: domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.safetyheads.akademiaandroida.domain.usecases
+
+
+import com.safetyheads.akademiaandroida.domain.repositories.UserRepository
+import kotlinx.coroutines.flow.Flow
+
+class UpdateProfileFCMUseCase(private val repository: UserRepository) :
```
Comment (msarelo) [inline domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt:7]:
```suggestion
class UpdateProfileFcmUseCase(private val repository: UserRepository) :
```

**Review by msarelo — CHANGES_REQUESTED** [review]

File: domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt (hunk 1 above)
Comment (rafalkrystek) [reply domain/src/main/java/com/safetyheads/akademiaandroida/domain/usecases/UpdateProfileFCMUseCase.kt:8]: This usecase requires UserUUID and FcmToken which are passed via UpdateProfileParam. If i use ParameterlessBaseUseCase i cannot pass any data

File: app/src/main/java/com/safetyheads/akademiaandroida/token/FirebaseTokenRepository.kt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+import com.google.firebase.messaging.FirebaseMessaging
+import com.safetyheads.akademiaandroida.domain.repositories.TokenRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.tasks.await
+
+class FirebaseTokenRepository: TokenRepository {
+    override fun getMessagingToken(): Flow<String> = flow {
+        val token = FirebaseMessaging.getInstance().token.await()
+        emit(token)
```
Comment (przemek19980102) [inline app/src/main/java/com/safetyheads/akademiaandroida/token/FirebaseTokenRepository.kt:12]: emit(FirebaseMessaging.getInstance().token.await()) wouldn't work?

File: data/src/main/java/com/safetyheads/akademiaandroida/data/network/repository/UserRepositoryImpl.kt [hunk 5]
```diff
@@ -36,6 +36,18 @@ class UserRepositoryImpl(
...
+    override suspend fun updateFcmToken(userUUID: String, token: String): Flow<Result<Unit>> = flow {
+        try {
+            FirebaseFirestore.getInstance().collection("users").document(userUUID).update(
+                mapOf("fcmToken" to token)
+            ).await()
+            emit(Result.success(Unit))
+        } catch (e: Exception) {
+            emit(Result.failure(e))
+        }
+
```
Comment (przemek19980102) [inline data/src/main/java/com/safetyheads/akademiaandroida/data/network/repository/UserRepositoryImpl.kt:48]: unnecessary empty line

**Review by przemek19980102 — APPROVED** [review]
Minor optional changes :)

File: app/src/main/java/com/safetyheads/akademiaandroida/token/FirebaseTokenRepository.kt (hunk 4 above)
Comment (forceindia712) [reply app/src/main/java/com/safetyheads/akademiaandroida/token/FirebaseTokenRepository.kt:12]: I would recommend making 'FirebaseMessaging.getInstance()' as Singleton in AndroidAcademyApplication

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/fragments/forgotpasswordfragment/user_test/UserTestViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    private val getMessagingTokenUseCase: GetMessagingTokenUseCase
+) : ViewModel() {
+
+    private var userUUID: String? = null
+    private val _message: MutableLiveData<String> = MutableLiveData<String>()
+    val message: LiveData<String> = _message
+
+    fun login() {
+        viewModelScope.launch {
+            loginUseCase.invoke(LoginUseCase.LoginParam("bagno1@test.pl", "pass1234")).collect { userUUIDResult ->
```
Comment (forceindia712) [inline presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/fragments/forgotpasswordfragment/user_test/UserTestViewModel.kt:28]: I don't think it's appropriate to include such data in the commit

**Pull request merged**